### PR TITLE
CLN: remove sqlalchemy<14 compat

### PR DESCRIPTION
**Pull request by fangchenli**

### CODE REVIEW

File: pandas/tests/io/test_sql.py [hunk 1]
```diff
@@ -1425,14 +1422,6 @@ def test_database_uri_string(self, test_frame1):
         tm.assert_frame_equal(test_frame1, test_frame3)
         tm.assert_frame_equal(test_frame1, test_frame4)
 
-    @td.skip_if_installed("pg8000")
```
Comment (mroeschke) [inline pandas/tests/io/test_sql.py:1428]: Why did this test need to be removed?

Comment (fangchenli) [reply pandas/tests/io/test_sql.py]: you are right. This doesn't belong here.

**Review by mroeschke — APPROVED** [review]
LGTM pending green

**Comment by jreback** [comment]
CI is bumped for >= 1.4? and there is a whatsnew note on this?

**Comment by fangchenli** [comment]
> CI is bumped for >= 1.4? and there is a whatsnew note on this?

yes #43191

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
thanks @fangchenli

**Pull request merged**